### ISSUE TITLE
about the missing rate

### ISSUE THREAD

**Issue by iTomxy** [XLearning-SCU/2021-CVPR-Completer#10]
Hi,
I wonder, why the missing rate is halved [here](https://github.com/XLearning-SCU/2021-CVPR-Completer/blob/main/get_mask.py#L17)?

**Comment by Lin-Yijie** [comment]
It is just a showcase. Our method could handle the incomplete problem of different missing rates and you can change it in configure.py.

**Comment by iTomxy** [comment]
As in the code, the missing rate is seemed to be determined in the [*get_mask*](https://github.com/XLearning-SCU/2021-CVPR-Completer/blob/main/get_mask.py) function.
However, if I comment out [this line](https://github.com/XLearning-SCU/2021-CVPR-Completer/blob/main/get_mask.py#L17), or change the [missing rate](https://github.com/XLearning-SCU/2021-CVPR-Completer/blob/main/configure.py#L17) to `1` in [*configure.py*](https://github.com/XLearning-SCU/2021-CVPR-Completer/blob/main/configure.py), the code will fail to reproduce the results under the incomplete setting (i.e. missing rate = `0.5`).

**Comment by Lin-Yijie** [comment]
The masked data is generated by the get_mask function and the corresponding missing rate is in configure.py. It should be pointed that the missing rate in configure.py is the real missing rate as talked about in the main paper. For the line missing_rate = missing_rate / 2, it is the process of generating incomplete multi-view datasets.
By the way, our method is trained on the complete multi-view data thus cannot handle the situation with missing rate=1.

**Comment by iTomxy** [comment]
Well, you're right as I compute the missing from the resultant mask and it's around `0.5`.
I guess you did that halving because there are `2` views and you want them each bear `0.25` missing rate. And I suggest using:
```python
missing_rate = missing_rate / view_num
```
which may be clearer.

**Comment by Lin-Yijie** [comment]
Thanks for your suggestions! I have changed the code to missing_rate = missing_rate / view_num now.

**Comment by zhangyuanyang21** [comment]
Hi, I would like to ask if running a dataset with the missing rate set to 0, is it a complete view network model?

**Comment by XLearning-SCU** [comment]
同学您好，我已经收到您的消息。                                    ——四川大学机器学习引论

**Comment by Lin-Yijie** [comment]
> Hi, I would like to ask if running a dataset with the missing rate set to 0, is it a complete view network model?

Yes it is~

**Comment by zhangyuanyang21** [comment]
Thank you very much for your reply,

**Comment by zhangyuanyang21** [comment]
你好，我发现Competer网络只能跑两个视图的数据集超过两个视图就不可以了，请问是不是这样的呢？

**Comment by Lin-Yijie** [comment]
Exactly. You could refer to our multi-view version [DCP](https://github.com/XLearning-SCU/2022-TPAMI-DCP).

**Comment by zhangyuanyang21** [comment]
Thank you very much for your reply！
